### ISSUE TITLE
[Kubernetes] Priority and same host ingresses not working

### ISSUE THREAD

**Issue by Sh4d1** [containous/traefik#3710]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
I have two ingresses like this:
```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: app
  annotations:
    kubernetes.io/ingress.class: traefik
    traefik.frontend.priority: "2"
spec:
  rules:
  - host: test.com
    http:
      paths:
      - path: /
        backend:
          serviceName: app
          servicePort: 80
```
and 
```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: maintenance
  annotations:
    kubernetes.io/ingress.class: traefik
    traefik.frontend.priority: "1"
spec:
  rules:
  - host: test.com
    http:
      paths:
      - path: /
        backend:
          serviceName: maintenance
          servicePort: 8080
```

### What did you expect to see?
I expected to see only the app backend used, and the mainteance one only when the app was down. 

### What did you see instead?
However, both are set up as backend for the same fronted, and doing a wrr load balancing.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.0-rc2                                                                                                                    
Codename:     maroilles
Go version:   go1.10.3
Built:        2018-07-17_12:42:38PM
OS/Arch:      linux/amd64
```

Also tested with

```
Version:      v1.6.5                                                                                                                        
Codename:     tetedemoine
Go version:   go1.10.3
Built:        2018-07-10_03:54:03PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http", "https"]
        
        [kubernetes]

        [web]
        address = ":8080"

        [entryPoints]
          [entryPoints.http]
          address = ":80"
            [entryPoints.http.redirect]
            entryPoint = "https"
          [entryPoints.https]
          address = ":443"
            [entryPoints.https.tls]
              [[entryPoints.https.tls.certificates]] 
              CertFile = "/ssl/tls.crt"
              KeyFile = "/ssl/tls.key"
```

Result of `/api`
```json
"test.com\/": {
        "servers": {
          "app-74bc77f9cd-6vjsl": {
            "url": "http:\/\/10.244.1.26:80",
            "weight": 1
          },
          "maintenance-698cf676d9-xlc8g": {
            "url": "http:\/\/10.244.1.38:8080",
            "weight": 1
          }
        },
        "loadBalancer": {
          "method": "wrr"
        }
      },
```

**Comment by Sh4d1** [comment]
Maybe an annotation specifying the backend name (default to host+path so it seems logical that two ingress with the same host and path would lead to the same backend) would help to fix the issue. I'll start to work on a PR tomorrow if I have some time.

**Comment by dtomcej** [comment]
The `baseName` property is only built with the Host and Path properties of the ingress:
https://github.com/containous/traefik/blob/v1.7/provider/kubernetes/kubernetes.go#L227

@timoreimann can you think of any issues with having `basename` being:

`{{namespace}}/{{ingress.name}}-{{host}}{{path}}`?

In theory you can't have a duplicate ingress with the same name in the same namespace...so that _should_ be unique enough right?

**Comment by timoreimann** [comment]
Isn't the annotation name just wrong? According to our [docs](https://docs.traefik.io/configuration/backends/kubernetes/#general-annotations), it should say `traefik.ingress.kubernetes.io/priority`.

**Comment by dtomcej** [comment]
It is caught by the compatibility check:
https://github.com/containous/traefik/blob/v1.7/provider/kubernetes/annotations.go#L74

**Comment by Sh4d1** [comment]
@timoreimann I tried with this annotation as well without success

**Comment by Sh4d1** [comment]
@dtomcej I confirm that the name you proposed is unique

**Comment by timoreimann** [comment]
Gotcha.

It should be fairly easy to repro the case and debug.

**Comment by traefiker** [comment]
Closed by #3743.
